### PR TITLE
Two fit dann

### PR DESCRIPTION
**Pull request by calanchue**
two fit dann example without switch and same in-out batch size.
i suggested at fchollet#4031
Evaluating target samples on DANN model Accuracy : 0.801582532051.
more accurate then original one : 0.767127403846
This is minimal modification version for concept proofing. If this implementation is good enough, we need to revert redundant modifications on core interface of keras.

### CODE REVIEW

**Comment by calanchue** [comment]
more revert for core and examples. As you can see in dann.py, I implemented same functionality without core interface modifications.

**Comment by calanchue** [comment]
Original Implementation Reversal layer can't be saved of loaded. I made a little bug fix.

**Comment by Pulkit-Khandelwal** [comment]
@calanchue @VanushVaswani Is this PR still active? I see that this was written around two years ago. I have used your code and made it compatible with the current Keras version. I have also used fit instead of train_on_batch. Moreover, instead of defining the updates of the learning rate and lambda, I have put them into the keras backend itself. That way the updates can happen at every iteration and the user doesn't have to deal with that. Do you want to open this again as a PR for Keras?

**Comment by ustcjerry** [comment]
> @calanchue @VanushVaswani Is this PR still active? I see that this was written around two years ago. I have used your code and made it compatible with the current Keras version. I have also used fit instead of train_on_batch. Moreover, instead of defining the updates of the learning rate and lambda, I have put them into the keras backend itself. That way the updates can happen at every iteration and the user doesn't have to deal with that. Do you want to open this again as a PR for Keras?

Though it has been a long time, I have been following this issue and pull request, would you like to share your code? @Pulkit-Khandelwal

**Comment by sreenivasaupadhyaya** [comment]
@Pulkit-Khandelwal Could you please  share your code here for the keras implementation of DANN or any open source code that works on keras backend?
